plugin_runner/plugin_runner.py

```diff
@@ -66,9 +66,15 @@ def __init__(self) -> None:
     async def HandleEvent(self, request: Event, context: Any) -> EventResponse:
         """This is invoked when an event comes in."""
         event_start_time = time.time()
-        event_name = EventType.Name(request.type)
+        event_type = request.type
+        event_name = EventType.Name(event_type)
         relevant_plugins = EVENT_PROTOCOL_MAP.get(event_name, [])
 
+        if event_type in [EventType.PLUGIN_CREATED, EventType.PLUGIN_UPDATED]:
+            plugin_name = request.target
+            # filter only for the plugin(s) that were created/updated
+            relevant_plugins = [p for p in relevant_plugins if p.startswith(f"{plugin_name}:")]
+
```

Comment:
@jrwils here is the logic where we only process the event for the plugin that was updated/changed